pkg/builder/builder.go

```diff
@@ -590,8 +624,33 @@ func overrideProbe(probe, ov *corev1.Probe) {
 		return
 	}
 	// Merge in parts of the override into the default probe
+	//
+	// You can only set one handler (exec, tcpSocket, httpGet or grpc). If the
+	// override has any one of those set, we always clear the other ones. A
+	// webhook exists that prevents setting more than one in the CR.
 	if ov.Exec != nil {
 		probe.Exec = ov.Exec
+		probe.TCPSocket = nil
```

Comment:
I get why you added this but why do you need to if we know that the default handler is `exec` and there is a webhook rule preventing more than one to be set in the CR. When a new probe is created aren't all handlers `nil` before setting?

Reply:
The main reason I did this was for the case where the default probe uses the `exec` handler, but we overwrite it in the CR with a `tcpSocket` handler. I ended up generalizing this for all handler as I didn't want to make assumptions about the default probe.